File: server/src/gameLogic/Enums/Regions.ts
import { Region } from '../Models/Region';
import { AreaName } from './AreaNames';

const eriadorAreas = [AreaName.FORLINDON, AreaName.MITHLOND, AreaName.HARLINDON, AreaName.TOWER_HILLS, AreaName.EVENDIM_HILLS, AreaName.LUNE_VALLEY, AreaName.THE_SHIRE];
const arnorAreas = [AreaName.FORODWAITH, AreaName.ANGMAR, AreaName.EASTERN_ANGMAR, AreaName.BORDERLANDS, AreaName.NORTH_DOWNS, AreaName.RHUDAUR, AreaName.FORNOST, AreaName.WEATHER_HILLS, AreaName.BUCKLAND, AreaName.OLD_FOREST, AreaName.SOUTH_DOWNS];
const rhunAreas = [AreaName.NORTH_RHUN, AreaName.WITHERED_HEATH, AreaName.ESGAROTH, AreaName.SOUTH_RHUN];
const mirkwoodAreas = [AreaName.CARROCK, AreaName.NORTH_MIRKWOOD, AreaName.ANDUIN_VALLEY, AreaName.EASTERN_MIRKWOOD, AreaName.SOUTH_MIRKWOOD];
const rohanAreas = [AreaName.EREGION, AreaName.MINHIRIATH, AreaName.DUNLAND, AreaName.ENEDWAITH, AreaName.FANGORN, AreaName.WEST_ROHAN, AreaName.GAP_OF_ROHAN];
const rhovanionAreas = [AreaName.MORIA, AreaName.GLADDEN_FIELDS, AreaName.LORIEN, AreaName.EMYN_MUIL, AreaName.BROWNLANDS, AreaName.RHUN_HILLS, AreaName.THE_WOLD, AreaName.DEAD_MARSHES];
const gondorAreas = [AreaName.DRUWAITH_IAUR, AreaName.ANDRAST, AreaName.ANFALAS, AreaName.VALE_OF_ERECH, AreaName.LAMEDON, AreaName.BELFALAS, AreaName.LEBENNIN, AreaName.MINAS_TIRITH, AreaName.ITHILIEN, AreaName.SOUTH_ITHILIEN];
const mordorAreas = [AreaName.UDUN_VALE, AreaName.MOUNT_DOOM, AreaName.BARAD_DUR, AreaName.MINAS_MORGUL, AreaName.GORGOROTH, AreaName.NURN];
const haradwaithAreas = [AreaName.HARONDOR, AreaName.UMBAR, AreaName.DEEP_HARAD, AreaName.HARAD, AreaName.NEAR_HARAD, AreaName.KHAND];

interface ListOfRegions { 
    [key: string]: Region
}

export const Regions: ListOfRegions = {
    ERIADOR: new Region('eriador', eriadorAreas, 3),
    ARNOR: new Region('arnor', arnorAreas, 7),
    RHUN: new Region('rhun', rhunAreas, 2),
    MIRKWOOD: new Region('mirkwood', mirkwoodAreas, 4),
    ROHAN: new Region('rohan', rohanAreas, 4),
    ROHVANION: new Region('rhovanion', rhovanionAreas, 4),
    GONDOR: new Region('gondor', gondorAreas, 7),
    MORDOR: new Region('mordor', mordorAreas, 2),
    HARADWAITH: new Region('haradwaith', haradwaithAreas, 2),
}

File: server/game_logic/enums/regions.py
from game_logic.models.Region import Region

eriador_area_names = [
    "FORLINDON",
    "MITHLOND",
    "HARLINDON",
    "TOWER_HILLS",
    "EVENDIM_HILLS",
    "LUNE_VALLEY",
    "THE_SHIRE"
]

arnor_area_names = [
    "FORODWAITH",
    "ANGMAR",
    "EASTERN_ANGMAR",
    "BORDERLANDS",
    "NORTH_DOWNS",
    "RHUDAUR",
    "FORNOST",
    "WEATHER_HILLS",
    "BUCKLAND",
    "OLD_FOREST",
    "SOUTH_DOWNS"
]

rhun_area_names = [
    "NORTH_RHUN",
    "WITHERED_HEATH",
    "ESGAROTH",
    "SOUTH_RHUN"
]

mirkwood_area_names = [
    "CARROCK",
    "NORTH_MIRKWOOD",
    "ANDUIN_VALLEY",
    "EASTERN_MIRKWOOD",
    "SOUTH_MIRKWOOD"
]

rohan_area_names = [
    "EREGION",
    "MINHIRIATH",
    "DUNLAND",
    "ENEDWAITH",
    "FANGORN",
    "WEST_ROHAN",
    "GAP_OF_ROHAN"
]

rhovanion_area_names = [
    "MORIA",
    "GLADDEN_FIELDS",
    "LORIEN",
    "EMYN_MUIL",
    "BROWNLANDS",
    "RHUN_HILLS",
    "THE_WOLD",
    "DEAD_MARSHES"
]

gondor_area_names = [
    "DRUWAITH_IAUR",
    "ANDRAST",
    "ANFALAS",
    "VALE_OF_ERECH",
    "LAMEDON",
    "BELFALAS",
    "LEBENNIN",
    "MINAS_TIRITH",
    "ITHILIEN",
    "SOUTH_ITHILIEN"
]

mordor_area_names = [
    "UDUN_VALE",
    "MOUNT_DOOM",
    "BARAD_DUR",
    "MINAS_MORGUL",
    "GORGOROTH",
    "NURN"
]

haradwaith_area_names = [
    "HARONDOR",
    "UMBAR",
    "DEEP_HARAD",
    "HARAD",
    "NEAR_HARAD",
    "KHAND"
]

regions = {
    "ERIADOR": Region("ERIADOR", eriador_area_names, 3),
    "ARNOR": Region("ARNOR", arnor_area_names, 7),
    "RHUN": Region("RHUN", rhun_area_names, 2),
    "MIRKWOOD": Region("MIRKWOOD", mirkwood_area_names, 4),
    "ROHAN": Region("ROHAN", rohan_area_names, 4),
    "RHOVANION": Region("RHOVANION", rhovanion_area_names, 4),
    "GONDOR": Region("GONDOR", gondor_area_names, 7),
    "MORDOR": Region("MORDOR", mordor_area_names, 2),
    "HARADWAITH": Region("HARADWAITH", haradwaith_area_names, 2),
}
